pkg/ddc/alluxio/shutdown.go

```diff
@@ -179,12 +181,73 @@ func (e *AlluxioEngine) destroyWorkers(workers int32) (err error) {
 	err = e.List(context.TODO(), nodeList, &client.ListOptions{
 		LabelSelector: datasetLabels,
 	})
+
 	if err != nil {
 		return
 	}
 
+	currentWorkers = int32(len(nodeList.Items))
+	if expectedWorkers >= currentWorkers {
+		e.Log.Info("No need to scale in. Skip.")
+		return currentWorkers, nil
+	}
+
+	var nodes []corev1.Node
+	if expectedWorkers >= 0 {
+		e.Log.V(1).Info("Scale in Alluxio workers", "expectedWorkers", expectedWorkers)
+		// This is a scale in operation
+		runtimeInfo, err := e.getRuntimeInfo()
+		if err != nil {
+			e.Log.Error(err, "getRuntimeInfo when scaling in")
+			return expectedWorkers, err
```

Comment:
If err, return currentWorkers is better?

Reply:
Thanks, Done